src/libs/NetworkConnection.ts

```diff
@@ -72,42 +73,51 @@ Onyx.connect({
  * internet connectivity or not. This is more reliable than the Pusher
  * `disconnected` event which takes about 10-15 seconds to emit.
  */
-function subscribeToNetInfo(): void {
-    // Note: We are disabling the configuration for NetInfo when using the local web API since requests can get stuck in a 'Pending' state and are not reliable indicators for "offline".
+function subscribeToNetInfo() {
+    let backendReachabilityCheckInterval: NodeJS.Timeout;
+    // Note: We are disabling the reachability check when using the local web API since requests can get stuck in a 'Pending' state and are not reliable indicators for "offline".
     // If you need to test the "recheck" feature then switch to the production API proxy server.
     if (!CONFIG.IS_USING_LOCAL_WEB) {
-        // Calling NetInfo.configure (re)checks current state. We use it to force a recheck whenever we (re)subscribe
-        NetInfo.configure({
-            // By default, NetInfo uses `/` for `reachabilityUrl`
-            // When App is served locally (or from Electron) this address is always reachable - even offline
+        // Set interval to (re)check current state every 60 seconds
+        backendReachabilityCheckInterval = setInterval(() => {
+            // Offline status also implies backend unreachability
+            if (isOffline) {
+                return;
+            }
             // Using the API url ensures reachability is tested over internet
-            reachabilityUrl: `${CONFIG.EXPENSIFY.DEFAULT_API_ROOT}api?command=Ping`,
-            reachabilityMethod: 'GET',
-            reachabilityTest: (response) => {
-                if (!response.ok) {
-                    return Promise.resolve(false);
-                }
-                return response
-                    .json()
-                    .then((json) => Promise.resolve(json.jsonCode === 200))
-                    .catch(() => Promise.resolve(false));
-            },
-
-            // If a check is taking longer than this time we're considered offline
-            reachabilityRequestTimeout: CONST.NETWORK.MAX_PENDING_TIME_MS,
-        });
+            fetch(`${CONFIG.EXPENSIFY.DEFAULT_API_ROOT}api?command=Ping`, {
+                method: 'GET',
+                cache: 'no-cache',
+            })
+                .then((response) => {
+                    if (!response.ok) {
+                        return Promise.resolve(false);
+                    }
+                    return response
+                        .json()
+                        .then((json) => Promise.resolve(json.jsonCode === 200))
+                        .catch(() => Promise.resolve(false));
+                })
+                .then(NetworkActions.setIsBackendReachable)
+                .catch(() => NetworkActions.setIsBackendReachable(false));
+        }, CONST.NETWORK.REACHABILITY_TIMEOUT_MS);
     }
 
     // Subscribe to the state change event via NetInfo so we can update
```

Comment:
I suggest changing this comment to...

> Using NetInfo for checking internet reachability is more accurate than the reliance on the Pusher `disconnected` event, which has a 10-15 second delay.

(In other words, I suggest moving some info from the `subscribeToNetInfo` function comment doc to this local doc)